scripts/populate_pool_batch_and_pools.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+from datetime import date
+
+from EPPs.common import StepEPP, InvalidStepError
+
+
+class PopulatePoolBatchPools(StepEPP):
+    # populate the pool ids with format YY-MM-DD_PDP_Pool#{1-9}{1,3} where {1-9}{1,3} is the next available number for
+    # that date.
+    # populate the pool batch ids with format YY-MM-DD_PDP_Batch#{1-9}{1,3} where {1-9}{1,3} is the next available number for
+    # that date.
+    # populate the output artifact with the prep workflow and adapter type of the input
+    _max_nb_output_containers = 1
+
+    def calculate_NTP_volume(self, pool_name):
+        inputs = self.process.step.pools.pooled_inputs[pool_name][1]
+
+        # obtain the Remaing Yield value from the submitted sample for each in put and find the maximum value
+        # the volume of NTP is based on all libraries in the pool gaining the same amount of yield as required by
+        # the library that requires the largest (maximum) yield i.e. all libraries will have the same volume of NTP to
+        # help minimise the problem uncertain sequencing yield caused by unbalanced pools
+        max_remaining_yield = max([input.samples[0].udf['Remaining Yield (Gb)'] for input in inputs])
+
+        # find the total amount of data that the pool will need to generate for all libraries combined
+        total_remaining_yield = int(max_remaining_yield) * len(inputs)
+
+        # find the total volume of library (NTP) that will be required from all libraries combined assuming 10 ul of pool
+        # is added to each CST well. 0.083 is a constant representing the amount of NTP required to generate 1 Gb of
+        # data in a balanced pool on one lane of a HiSeqX flowcell
+        total_volume_of_ntp_for_yield = total_remaining_yield * 0.083
+
+        #find the number of lanes required, assuming that 10 of NTP is used in each CST well and only whole wells can
+        #be used
+        number_of_lanes = -(-total_volume_of_ntp_for_yield//10) #using // always rounds down so if we make the
+        #numerator negative then rounding down is effectively rounding up if we then make the answer positive by adding
+        #another negative
```

Comment:
I understand the goal here from the comment but the code could be made less cryptic by using the function [ceil](https://docs.python.org/3.6/library/math.html#math.ceil) from math module:
```python
        number_of_lanes = math.ceil(total_volume_of_ntp_for_yield / 10)
```

Reply:
updated